pkg/clusters/addons/kong/addon.go

```diff
@@ -394,6 +402,79 @@ func (a *Addon) DumpDiagnostics(ctx context.Context, cluster clusters.Cluster) (
 
 	switch a.proxyDBMode {
 	case PostgreSQL:
+		out, err := os.CreateTemp(os.TempDir(), "ktf-kong-")
+		if err != nil {
+			return diagnostics, fmt.Errorf("could not create temp file: %w", err)
+		}
+		defer os.Remove(out.Name())
+		dumpConfig := dump.Config{}
+		addr, err := a.ProxyAdminURL(ctx, cluster)
+		if err != nil {
+			return diagnostics, fmt.Errorf("could not build Kong client: %w", err)
+		}
+		opts := deckutils.KongClientConfig{
+			Address: addr.String(),
+			HTTPClient: &http.Client{
+				Timeout: time.Second * 90, //nolint:gomnd
+			},
+			TLSSkipVerify: true,
+		}
+		if a.proxyEnterpriseSuperAdminPassword != "" {
+			opts.Headers = append(opts.Headers, "kong-admin-token:"+a.proxyEnterpriseSuperAdminPassword)
+		}
+		client, err := deckutils.GetKongClient(opts)
+		if err != nil {
+			return diagnostics, fmt.Errorf("could not build Kong client: %w", err)
+		}
+		workspaces, err := client.Workspaces.ListAll(ctx)
+		var kongAPIError *kong.APIError
+		if errors.As(err, &kongAPIError) && kongAPIError.Code() == http.StatusNotFound {
+			defaultws := kong.Workspace{Name: kong.String("default")}
+			workspaces = append(workspaces, &defaultws)
```

Comment:
Am I getting this right that this relies on the fact that if the `err` is non `nil` then  `workspaces` are empty hence we can `append` on it and get a slice with 1 element? If so I'd urge us no to do so, but just assign `workspaces` to a slice with one element: `defaultws`. WDYT?